cylc/flow/network/ssh_client.py

```diff
@@ -45,11 +47,43 @@ def __init__(
             timeout: Union[float, str] = None
     ):
         self.workflow = workflow
-
+        self.SLEEP_INTERVAL = 0.1
         if not host:
             self.host, _, _ = get_location(workflow)
 
-    def send_request(self, command, args=None, timeout=None):
+    async def async_request(self, command, args=None, timeout=None):
+        """Send asynchronous request via SSH.
+        """
+        self.timeout = timeout if timeout else 300  # 5 min default timeout
+        try:
+            async with ascyncto(self.timeout):
+                cmd, ssh_cmd, login_sh, cylc_path, msg = self.prepare_command(
+                    command, args, timeout)
+                proc = _remote_cylc_cmd(
+                    cmd,
+                    host=self.host,
+                    stdin_str=msg,
+                    ssh_cmd=ssh_cmd,
+                    remote_cylc_path=cylc_path,
+                    ssh_login_shell=login_sh,
+                    capture_process=True)
+                while True:
+                    if proc.poll() is not None:
+                        break
+                    await asyncio.sleep(self.SLEEP_INTERVAL)
+                    out, err = (f.decode() for f in proc.communicate())
+                    return_code = proc.wait()
+                    if return_code:
+                        raise ClientError(err, f"return-code={return_code}")
```

Comment:
Both `Popen.communicate()` and `Popen.wait()` wait for the subprocess to finish, so that makes the while loop unnecessary. I'm not convinced this is a genuine async method (but does that matter?)

Reply:
I think this was an indentation error: d82212ab3832c9330ec44ead007f80e15cc30007